### PR TITLE
Bug in sketch on face

### PR DESCRIPTION
**Pull request by adamchalmers**
This should add a second plumbus, but instead it fails because the tagged circle "arc-b" cannot be found.

The cause is that the API call to `Solid3dGetExtrusionFaceInfo` returns 3 `ExtrusionFaceInfo`s when creating the first plumbus, but 0 for the second plumbus. 

once fixed will close https://github.com/KittyCAD/modeling-app/issues/1894

### CODE REVIEW

**Comment by adamchalmers** [comment]
I'm not sure this is an engine bug -- it could be, but, this was working before when the modeling-app was double-executing the first command of each pipeline. I don't know why fixing that would break this.

**Comment by Irev-Dev** [comment]
In case it's useful here's a log of the plumbus calls

<details>
<summary>

logs

</summary>

```JSON
[
  {
    "type": "send-scene",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "make_plane",
        "size": 100,
        "origin": {
          "x": 0,
          "y": 0,
          "z": 0
        },
        "x_axis": {
          "x": 1,
          "y": 0,
          "z": 0
        },
        "y_axis": {
          "x": 0,
          "y": 1,
          "z": 0
        },
        "clobber": false,
        "hide": true
      },
      "cmd_id": "d2a7f2f2-25a5-487a-89a9-665fba5a589d"
    }
  },
  {
    "type": "send-modeling",
    "data": {
      "type": "modeling_cmd_batch_req",
      "batch_id": "ace3a401-5186-4e47-bd76-1ee0882206d6",
      "requests": [
        {
          "cmd": {
            "type": "set_scene_units",
            "unit": "mm"
          },
          "cmd_id": "a8a3eff2-f6e6-4f8d-9bbc-1f5549f22057"
        },
        {
          "cmd": {
            "type": "make_plane",
            "clobber": false,
            "hide": true,
            "origin": {
              "x": 0,
              "y": 0,
              "z": 0
            },
            "size": 60,
            "x_axis": {
              "x": 1,
              "y": 0,
              "z": 0
            },
            "y_axis": {
              "x": 0,
              "y": 1,
              "z": 0
            }
          },
          "cmd_id": "9a9fd88a-5a55-4b82-b712-6704f3966be7"
        },
        {
          "cmd": {
            "type": "sketch_mode_enable",
            "animated": false,
            "disable_camera_with_plane": {
              "x": 0,
              "y": 0,
              "z": 1
            },
            "ortho": false,
            "plane_id": "9a9fd88a-5a55-4b82-b712-6704f3966be7"
          },
          "cmd_id": "2a54febb-2b19-4c7f-b5e5-0377a6fbbcc6"
        },
        {
          "cmd": {
            "type": "start_path"
          },
          "cmd_id": "2a1b652a-0055-4f02-9228-d6f57af67708"
        },
        {
          "cmd": {
            "type": "move_path_pen",
            "path": "2a1b652a-0055-4f02-9228-d6f57af67708",
            "to": {
              "x": -24,
              "y": -24,
              "z": 0
            }
          },
          "cmd_id": "8801596d-e403-448d-bb62-99ca0ce85294"
        },
        {
          "cmd": {
            "type": "extend_path",
            "path": "2a1b652a-0055-4f02-9228-d6f57af67708",
            "segment": {
              "type": "line",
              "end": {
                "x": 48,
                "y": 0,
                "z": 0
              },
              "relative": true
            }
          },
          "cmd_id": "8ed4e058-2d37-4bfa-9b6f-83f6a9404f75"
        },
        {
          "cmd": {
            "type": "extend_path",
            "path": "2a1b652a-0055-4f02-9228-d6f57af67708",
            "segment": {
              "type": "line",
              "end": {
                "x": 14.832815729997478,
                "y": 45.65071278216737,
                "z": 0
              },
              "relative": true
            }
          },
          "cmd_id": "e02f57a5-7e9f-4fb9-9083-6603ff51efa3"
        },
        {
          "cmd": {
            "type": "extend_path",
            "path": "2a1b652a-0055-4f02-9228-d6f57af67708",
            "segment": {
              "type": "line",
              "end": {
                "x": -38.83281572999747,
                "y": 28.213692110038714,
                "z": 0
              },
              "relative": true
            }
          },
          "cmd_id": "93f7a91c-e8c5-4a7c-9717-df7fddff0caf"
        },
        {
          "cmd": {
            "type": "extend_path",
            "path": "2a1b652a-0055-4f02-9228-d6f57af67708",
            "segment": {
              "type": "line",
              "end": {
                "x": -38.83281572999748,
                "y": -28.213692110038707,
                "z": 0
              },
              "relative": true
            }
          },
          "cmd_id": "5bd707ef-f45a-41ed-894c-ec50f2b5fa0a"
        },
        {
          "cmd": {
            "type": "extend_path",
            "path": "2a1b652a-0055-4f02-9228-d6f57af67708",
            "segment": {
              "type": "line",
              "end": {
                "x": 14.832815729997478,
                "y": -45.65071278216737,
                "z": 0
              },
              "relative": true
            }
          },
          "cmd_id": "93c72490-f738-49f5-a5c4-ba6343cfc35f"
        }
      ]
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "1c1b5c54-926f-4f9c-a956-7bf190a1fe4c"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "d2a7f2f2-25a5-487a-89a9-665fba5a589d"
  },
  {
    "type": "send-scene",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "plane_set_color",
        "plane_id": "d2a7f2f2-25a5-487a-89a9-665fba5a589d",
        "color": {
          "r": 0.7,
          "g": 0.28,
          "b": 0.28,
          "a": 0.4
        }
      },
      "cmd_id": "3186d030-24c8-41d0-8d86-f62082b64d3e"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "a8a3eff2-f6e6-4f8d-9bbc-1f5549f22057",
    "cmd_type": "set_scene_units"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "9a9fd88a-5a55-4b82-b712-6704f3966be7",
    "cmd_type": "make_plane"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "2a54febb-2b19-4c7f-b5e5-0377a6fbbcc6",
    "cmd_type": "sketch_mode_enable"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "2a1b652a-0055-4f02-9228-d6f57af67708",
    "cmd_type": "start_path"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "8801596d-e403-448d-bb62-99ca0ce85294",
    "cmd_type": "move_path_pen"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "8ed4e058-2d37-4bfa-9b6f-83f6a9404f75",
    "cmd_type": "extend_path"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "e02f57a5-7e9f-4fb9-9083-6603ff51efa3",
    "cmd_type": "extend_path"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "93f7a91c-e8c5-4a7c-9717-df7fddff0caf",
    "cmd_type": "extend_path"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "5bd707ef-f45a-41ed-894c-ec50f2b5fa0a",
    "cmd_type": "extend_path"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "93c72490-f738-49f5-a5c4-ba6343cfc35f",
    "cmd_type": "extend_path"
  },
  {
    "type": "send-modeling",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "extrude",
        "cap": true,
        "distance": 30,
        "target": "2a1b652a-0055-4f02-9228-d6f57af67708"
      },
      "cmd_id": "403a0f5b-5e62-419d-a290-af8bf2d4fd32"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "3186d030-24c8-41d0-8d86-f62082b64d3e"
  },
  {
    "type": "send-scene",
    "data": {
      "type": "modeling_cmd_req",
      "cmd_id": "f157fd96-65d5-45c6-97c9-ac25b3055472",
      "cmd": {
        "type": "object_visible",
        "object_id": "d2a7f2f2-25a5-487a-89a9-665fba5a589d",
        "hidden": true
      }
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "403a0f5b-5e62-419d-a290-af8bf2d4fd32",
    "cmd_type": "extrude"
  },
  {
    "type": "send-modeling",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "object_bring_to_front",
        "object_id": "2a1b652a-0055-4f02-9228-d6f57af67708"
      },
      "cmd_id": "204a8fd0-611d-4482-af1c-f42763462a7f"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "f157fd96-65d5-45c6-97c9-ac25b3055472"
  },
  {
    "type": "send-scene",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "make_plane",
        "size": 100,
        "origin": {
          "x": 0,
          "y": 0,
          "z": 0
        },
        "x_axis": {
          "x": 0,
          "y": 1,
          "z": 0
        },
        "y_axis": {
          "x": 0,
          "y": 0,
          "z": 1
        },
        "clobber": false,
        "hide": true
      },
      "cmd_id": "5bbec679-256f-4172-8c99-aa8a11800f8e"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "204a8fd0-611d-4482-af1c-f42763462a7f",
    "cmd_type": "object_bring_to_front"
  },
  {
    "type": "send-modeling",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "solid3d_get_extrusion_face_info",
        "edge_id": "8ed4e058-2d37-4bfa-9b6f-83f6a9404f75",
        "object_id": "2a1b652a-0055-4f02-9228-d6f57af67708"
      },
      "cmd_id": "403a0f5b-5e62-419d-a290-af8bf2d4fd32"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "5bbec679-256f-4172-8c99-aa8a11800f8e"
  },
  {
    "type": "send-scene",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "plane_set_color",
        "plane_id": "5bbec679-256f-4172-8c99-aa8a11800f8e",
        "color": {
          "r": 0.28,
          "g": 0.7,
          "b": 0.28,
          "a": 0.4
        }
      },
      "cmd_id": "205a8107-7087-489f-aee4-1b0cb3142b3f"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "solid3d_get_extrusion_face_info",
          "data": {
            "faces": [
              {
                "curve_id": "8ed4e058-2d37-4bfa-9b6f-83f6a9404f75",
                "face_id": "97cfc06f-67b0-48f1-8936-0cf7daa92c16",
                "cap": "none"
              },
              {
                "curve_id": "e02f57a5-7e9f-4fb9-9083-6603ff51efa3",
                "face_id": "b3d4ce58-ebf2-42b4-bea5-252f12a1586a",
                "cap": "none"
              },
              {
                "curve_id": "93f7a91c-e8c5-4a7c-9717-df7fddff0caf",
                "face_id": "85bcea3b-5da4-44f5-95d1-4f9a81a6191f",
                "cap": "none"
              },
              {
                "curve_id": "5bd707ef-f45a-41ed-894c-ec50f2b5fa0a",
                "face_id": "12290430-cc9c-45b7-ab4a-4a488f86852d",
                "cap": "none"
              },
              {
                "curve_id": "93c72490-f738-49f5-a5c4-ba6343cfc35f",
                "face_id": "400b7502-f2ce-4730-af4f-8deae7d6b49f",
                "cap": "none"
              },
              {
                "curve_id": null,
                "face_id": "188ca872-41d8-4b1c-8b59-36c9f3e74462",
                "cap": "bottom"
              },
              {
                "curve_id": null,
                "face_id": "0865c672-883c-40de-8406-c0e411f5af1f",
                "cap": "top"
              }
            ]
          }
        }
      }
    },
    "id": "403a0f5b-5e62-419d-a290-af8bf2d4fd32",
    "cmd_type": "solid3d_get_extrusion_face_info"
  },
  {
    "type": "send-modeling",
    "data": {
      "type": "modeling_cmd_batch_req",
      "batch_id": "29621747-94e2-493c-8a22-36994b297e67",
      "requests": [
        {
          "cmd": {
            "type": "enable_sketch_mode",
            "adjust_camera": false,
            "animated": false,
            "entity_id": "97cfc06f-67b0-48f1-8936-0cf7daa92c16",
            "ortho": false
          },
          "cmd_id": "6a296303-167f-4607-8f45-f629e21b9833"
        },
        {
          "cmd": {
            "type": "start_path"
          },
          "cmd_id": "a0430597-c0fc-4669-ac78-1f077da6a6c7"
        },
        {
          "cmd": {
            "type": "move_path_pen",
            "path": "a0430597-c0fc-4669-ac78-1f077da6a6c7",
            "to": {
              "x": 9,
              "y": 0,
              "z": 0
            }
          },
          "cmd_id": "0e3a9a3d-646a-4d3c-a289-08fec593cba7"
        },
        {
          "cmd": {
            "type": "extend_path",
            "path": "a0430597-c0fc-4669-ac78-1f077da6a6c7",
            "segment": {
              "type": "arc",
              "center": {
                "x": 0,
                "y": 0
              },
              "end": {
                "unit": "degrees",
                "value": 360
              },
              "radius": 9,
              "relative": false,
              "start": {
                "unit": "degrees",
                "value": 0
              }
            }
          },
          "cmd_id": "bd481e16-2f90-48dd-b6d6-c3132869e05d"
        },
        {
          "cmd": {
            "type": "close_path",
            "path_id": "a0430597-c0fc-4669-ac78-1f077da6a6c7"
          },
          "cmd_id": "fc7a2f14-ef42-4c58-90bc-364bc391accb"
        }
      ]
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "205a8107-7087-489f-aee4-1b0cb3142b3f"
  },
  {
    "type": "send-scene",
    "data": {
      "type": "modeling_cmd_req",
      "cmd_id": "d569904a-1efa-4648-ae89-654bf9ff1504",
      "cmd": {
        "type": "object_visible",
        "object_id": "5bbec679-256f-4172-8c99-aa8a11800f8e",
        "hidden": true
      }
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "6a296303-167f-4607-8f45-f629e21b9833",
    "cmd_type": "enable_sketch_mode"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "a0430597-c0fc-4669-ac78-1f077da6a6c7",
    "cmd_type": "start_path"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "0e3a9a3d-646a-4d3c-a289-08fec593cba7",
    "cmd_type": "move_path_pen"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "bd481e16-2f90-48dd-b6d6-c3132869e05d",
    "cmd_type": "extend_path"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "fc7a2f14-ef42-4c58-90bc-364bc391accb",
    "cmd_type": "close_path"
  },
  {
    "type": "send-modeling",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "extrude",
        "cap": true,
        "distance": 18,
        "target": "a0430597-c0fc-4669-ac78-1f077da6a6c7"
      },
      "cmd_id": "3fd354ea-7160-4345-821d-2cbf6ca778e2"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "d569904a-1efa-4648-ae89-654bf9ff1504"
  },
  {
    "type": "send-scene",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "make_plane",
        "size": 100,
        "origin": {
          "x": 0,
          "y": 0,
          "z": 0
        },
        "x_axis": {
          "x": 1,
          "y": 0,
          "z": 0
        },
        "y_axis": {
          "x": 0,
          "y": 0,
          "z": 1
        },
        "clobber": false,
        "hide": true
      },
      "cmd_id": "f55b4e72-4bac-4909-9cf3-c8ebbef976e5"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "3fd354ea-7160-4345-821d-2cbf6ca778e2",
    "cmd_type": "extrude"
  },
  {
    "type": "send-modeling",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "sketch_mode_disable"
      },
      "cmd_id": "cdf39649-7f90-4a30-b68a-bf0079941954"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "f55b4e72-4bac-4909-9cf3-c8ebbef976e5"
  },
  {
    "type": "send-scene",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "plane_set_color",
        "plane_id": "f55b4e72-4bac-4909-9cf3-c8ebbef976e5",
        "color": {
          "r": 0.28,
          "g": 0.28,
          "b": 0.7,
          "a": 0.4
        }
      },
      "cmd_id": "4b1d5228-5c08-4ee4-abb6-52506f2bd7c7"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "cdf39649-7f90-4a30-b68a-bf0079941954",
    "cmd_type": "sketch_mode_disable"
  },
  {
    "type": "send-modeling",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "object_bring_to_front",
        "object_id": "a0430597-c0fc-4669-ac78-1f077da6a6c7"
      },
      "cmd_id": "8605d17b-25c4-4949-abee-ca1e4864d7d6"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "4b1d5228-5c08-4ee4-abb6-52506f2bd7c7"
  },
  {
    "type": "send-scene",
    "data": {
      "type": "modeling_cmd_req",
      "cmd_id": "e1266fe4-b257-4eed-ae73-bc8a7724dda0",
      "cmd": {
        "type": "object_visible",
        "object_id": "f55b4e72-4bac-4909-9cf3-c8ebbef976e5",
        "hidden": true
      }
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "8605d17b-25c4-4949-abee-ca1e4864d7d6",
    "cmd_type": "object_bring_to_front"
  },
  {
    "type": "send-modeling",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "solid3d_get_extrusion_face_info",
        "edge_id": "bd481e16-2f90-48dd-b6d6-c3132869e05d",
        "object_id": "2a1b652a-0055-4f02-9228-d6f57af67708"
      },
      "cmd_id": "3fd354ea-7160-4345-821d-2cbf6ca778e2"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "e1266fe4-b257-4eed-ae73-bc8a7724dda0"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "solid3d_get_extrusion_face_info",
          "data": {
            "faces": [
              {
                "curve_id": "bd481e16-2f90-48dd-b6d6-c3132869e05d",
                "face_id": "5e55b91f-2dd0-4721-853c-6266f25b7841",
                "cap": "none"
              },
              {
                "curve_id": null,
                "face_id": "cf919eee-cf32-40d0-8ea0-3a2f7525299b",
                "cap": "bottom"
              },
              {
                "curve_id": null,
                "face_id": "ac238a2d-65b5-4e71-af24-123a2cb89aaa",
                "cap": "top"
              }
            ]
          }
        }
      }
    },
    "id": "3fd354ea-7160-4345-821d-2cbf6ca778e2",
    "cmd_type": "solid3d_get_extrusion_face_info"
  },
  {
    "type": "send-modeling",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "solid3d_get_opposite_edge",
        "edge_id": "bd481e16-2f90-48dd-b6d6-c3132869e05d",
        "face_id": "5e55b91f-2dd0-4721-853c-6266f25b7841",
        "object_id": "2a1b652a-0055-4f02-9228-d6f57af67708"
      },
      "cmd_id": "cb8e522d-ccf9-4116-b76d-43f42f165d09"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "solid3d_get_opposite_edge",
          "data": {
            "edge": "3d3da7f3-a400-4186-90e3-a97ccaf14530"
          }
        }
      }
    },
    "id": "cb8e522d-ccf9-4116-b76d-43f42f165d09",
    "cmd_type": "solid3d_get_opposite_edge"
  },
  {
    "type": "send-modeling",
    "data": {
      "type": "modeling_cmd_batch_req",
      "batch_id": "ded913a3-44bf-420d-9803-7a6e58bf063a",
      "requests": [
        {
          "cmd": {
            "type": "solid3d_fillet_edge",
            "edge_id": "bd481e16-2f90-48dd-b6d6-c3132869e05d",
            "object_id": "2a1b652a-0055-4f02-9228-d6f57af67708",
            "radius": 0.4,
            "tolerance": 1e-7
          },
          "cmd_id": "8b99e155-9ace-4074-8864-71652786bf84"
        },
        {
          "cmd": {
            "type": "solid3d_fillet_edge",
            "edge_id": "3d3da7f3-a400-4186-90e3-a97ccaf14530",
            "object_id": "2a1b652a-0055-4f02-9228-d6f57af67708",
            "radius": 0.4,
            "tolerance": 1e-7
          },
          "cmd_id": "18cf2756-0872-46bb-a58f-b06dc1be6bf2"
        },
        {
          "cmd": {
            "type": "enable_sketch_mode",
            "adjust_camera": false,
            "animated": false,
            "entity_id": "b3d4ce58-ebf2-42b4-bea5-252f12a1586a",
            "ortho": false
          },
          "cmd_id": "4c2e42fe-dc6c-44ae-b561-b1aeecb52099"
        },
        {
          "cmd": {
            "type": "start_path"
          },
          "cmd_id": "6ea78590-5882-454a-bed4-b7fa72c4ed6a"
        },
        {
          "cmd": {
            "type": "move_path_pen",
            "path": "6ea78590-5882-454a-bed4-b7fa72c4ed6a",
            "to": {
              "x": 9,
              "y": 0,
              "z": 0
            }
          },
          "cmd_id": "4adc8f34-426d-49d7-8815-70bf88b0a726"
        },
        {
          "cmd": {
            "type": "extend_path",
            "path": "6ea78590-5882-454a-bed4-b7fa72c4ed6a",
            "segment": {
              "type": "arc",
              "center": {
                "x": 0,
                "y": 0
              },
              "end": {
                "unit": "degrees",
                "value": 360
              },
              "radius": 9,
              "relative": false,
              "start": {
                "unit": "degrees",
                "value": 0
              }
            }
          },
          "cmd_id": "0f72afaa-2aa4-4dd4-bf8f-43dc3d21fa8c"
        },
        {
          "cmd": {
            "type": "close_path",
            "path_id": "6ea78590-5882-454a-bed4-b7fa72c4ed6a"
          },
          "cmd_id": "cc919440-bc02-481d-b7ab-b5484e6fd373"
        }
      ]
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "8b99e155-9ace-4074-8864-71652786bf84",
    "cmd_type": "solid3d_fillet_edge"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "18cf2756-0872-46bb-a58f-b06dc1be6bf2",
    "cmd_type": "solid3d_fillet_edge"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "4c2e42fe-dc6c-44ae-b561-b1aeecb52099",
    "cmd_type": "enable_sketch_mode"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "6ea78590-5882-454a-bed4-b7fa72c4ed6a",
    "cmd_type": "start_path"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "4adc8f34-426d-49d7-8815-70bf88b0a726",
    "cmd_type": "move_path_pen"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "0f72afaa-2aa4-4dd4-bf8f-43dc3d21fa8c",
    "cmd_type": "extend_path"
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "cc919440-bc02-481d-b7ab-b5484e6fd373",
    "cmd_type": "close_path"
  },
  {
    "type": "send-modeling",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "extrude",
        "cap": true,
        "distance": 18,
        "target": "6ea78590-5882-454a-bed4-b7fa72c4ed6a"
      },
      "cmd_id": "eedd7a0a-783e-4d15-96e0-a9d8df4993d8"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "eedd7a0a-783e-4d15-96e0-a9d8df4993d8",
    "cmd_type": "extrude"
  },
  {
    "type": "send-modeling",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "sketch_mode_disable"
      },
      "cmd_id": "7bd6e8ee-36f5-4336-b64b-72a5634d4a81"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "7bd6e8ee-36f5-4336-b64b-72a5634d4a81",
    "cmd_type": "sketch_mode_disable"
  },
  {
    "type": "send-modeling",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "object_bring_to_front",
        "object_id": "6ea78590-5882-454a-bed4-b7fa72c4ed6a"
      },
      "cmd_id": "7ea58e34-f100-4d4c-a8b5-17189b22c9cb"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "empty"
        }
      }
    },
    "id": "7ea58e34-f100-4d4c-a8b5-17189b22c9cb",
    "cmd_type": "object_bring_to_front"
  },
  {
    "type": "send-modeling",
    "data": {
      "type": "modeling_cmd_req",
      "cmd": {
        "type": "solid3d_get_extrusion_face_info",
        "edge_id": "0f72afaa-2aa4-4dd4-bf8f-43dc3d21fa8c",
        "object_id": "2a1b652a-0055-4f02-9228-d6f57af67708"
      },
      "cmd_id": "eedd7a0a-783e-4d15-96e0-a9d8df4993d8"
    }
  },
  {
    "type": "receive-reliable",
    "data": {
      "type": "modeling",
      "data": {
        "modeling_response": {
          "type": "solid3d_get_extrusion_face_info",
          "data": {
            "faces": []
          }
        }
      }
    },
    "id": "eedd7a0a-783e-4d15-96e0-a9d8df4993d8",
    "cmd_type": "solid3d_get_extrusion_face_info"
  },
  {
    "type": "execution-done",
    "data": null
  }
]
```


</details>

**Comment by Irev-Dev** [comment]
https://github.com/KittyCAD/modeling-app/assets/29681384/ee0bb95e-2d75-46aa-ae91-efa98b2b51eb

**Comment by adamchalmers** [comment]
Oh that's an interesting find, thank you Kurt.

**Comment by mlfarrell** [comment]
I just recorded a long video going through all possible orders of filleting vs extrusions on face and it was too long to upload to GH.

TL;DR - the engine isn't invalidating ids and is handling things correctly no matter the order of how the extrusions/fillets are done, so keep checking into the modeling-app/executor for the cause of this issue.

**Comment by jessfraz** [comment]
I could be like we are exiting sketch mode or something subtle

**Comment by adamchalmers** [comment]
I've tried to copy the API calls that the modeling app makes, and turn them into an engine unit test here: https://github.com/KittyCAD/engine/pull/1977

Weirdly, it works fine in the engine test. So I either copied the calls wrong, omitting the modeling-app's mistake, or there's something weird going on beyond just the API calls.

**Comment by adamchalmers** [comment]
OK, I think this is a bug related to defining sketches in functions. The plumbuses can be filletted just fine as long as the `make_circle` function is inlined.


https://github.com/KittyCAD/modeling-app/assets/5407457/70adb938-2d28-4155-b59d-800d5f414e67

**Comment by adamchalmers** [comment]
https://github.com/KittyCAD/modeling-app/assets/5407457/bf2c147f-01b3-4bd3-ab6d-64a8e8bacd40

**Comment by adamchalmers** [comment]
Fixed in https://github.com/KittyCAD/modeling-app/pull/2474
